### PR TITLE
Elpi: fix catastrophic backtracking

### LINKED ISSUE #2053: Elpi lexer has pathological backtracking
The Elpi lexer has pathological backtracking that can be triggered with the following minimal input:

```python
from pygments.lexers.elpi import ElpiLexer

list(ElpiLexer().get_tokens("a" * 30))
```

Only lowercase characters trigger pathological backtracking. Uppercase characters and digits will not.

The culprit is at [line 62](https://github.com/pygments/pygments/blob/a502df0a3ebcffb64125025a5e8a3452d8c63a17/pygments/lexers/elpi.py#L62).

@gares I'm not familiar with Elpi, so if you can jump in to address this, great! Otherwise I'll be able to take a crack at it after this weekend.

### PR DESCRIPTION
**Pull request by Jean-Abou-Samra**
Closes #2053

### CODE REVIEW

File: pygments/lexers/elpi.py [hunk 1]
```diff
@@ -2,142 +2,172 @@
...
-    idcharstarns_re = r"({}+|(?=\.[a-z])\.{}+)".format(idchar_re,idchar_re)
-    symbchar_re = r"({}|{}|{}|{}|:)".format(lcase_re, ucase_re, digit_re, schar_re)
-    constant_re = r"({}{}*|{}{}*|{}{}*|_{}+)".format(ucase_re, idchar_re, lcase_re, idcharstarns_re,schar2_re, symbchar_re,idchar_re)
-    symbol_re=r"(,|<=>|->|:-|;|\?-|->|&|=>|\bas\b|\buvar\b|<|=<|=|==|>=|>|\bi<|\bi=<|\bi>=|\bi>|\bis\b|\br<|\br=<|\br>=|\br>|\bs<|\bs=<|\bs>=|\bs>|@|::|\[\]|`->|`:|`:=|\^|-|\+|\bi-|\bi\+|r-|r\+|/|\*|\bdiv\b|\bi\*|\bmod\b|\br\*|~|\bi~|\br~)"
-    escape_re=r"\(({}|{})\)".format(constant_re,symbol_re)
-    const_sym_re = r"({}|{}|{})".format(constant_re,symbol_re,escape_re)
+    schar_re = rf"({schar2_re}|-|\$|_)"
+    idchar_re = rf"([a-zA-Z]|{digit_re}|{schar_re})"
+    symbchar_re = rf"({idchar_re}|:)"
+    constant_re = rf"([A-Z]{idchar_re}*|[a-z]{idchar_re}*(\.[a-zA-Z]{idchar_re}*)*|{schar2_re}{symbchar_re}*|_{idchar_re}+)"
```
Comment (Jean-Abou-Samra) [inline pygments/lexers/elpi.py:34]: Note that this doesn't allow dots when the constant begins with a capital letter. I assume that is correct as it was already the case before.

Comment (gares) [reply pygments/lexers/elpi.py:34]: it is

**Review by gares — APPROVED** [review]

**Review by birkenfeld — CHANGES_REQUESTED** [review]
The reformatting makes it basically impossible to see what has really changed.

Please don't apply `black` to the Pygments codebase, as you can see we have a pretty different style.

**Comment by Jean-Abou-Samra** [comment]
> The reformatting makes it basically impossible to see what has really changed.
>
> Please don't apply black to the Pygments codebase, as you can see we have a pretty different style.

OK, I'll back the formatting out. Note, though, that it is in a separate commit for the purpose of review (I've been trained rigorously to use atomic commits, but maybe that is not usual here?).

**Comment by birkenfeld** [comment]
As far as I can see, the formatting changes in the first commit ("minor changes") which also has a bunch of other (useful) things?

**Comment by birkenfeld** [comment]
> I've been trained rigorously to use atomic commits, but maybe that is not usual here?

Of course it's useful, but even with atomic commits I wouldn't expect to see a wholesale reformatting in a PR to fix a bug :-)

**Comment by Jean-Abou-Samra** [comment]
I have refactored this PR to include only the bugfix.

**Review by birkenfeld — APPROVED** [review]
Thanks a lot!

**Pull request merged**